### PR TITLE
meta: improve Utils.defaults typing and fix test

### PR DESCRIPTION
**Pull request by WikiRik**
## Pull Request Checklist

- [X] Have you added new tests to prevent regressions?
- [X] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [X] Did you update the typescript typings accordingly (if applicable)?
- [X] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [X] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR is built on #15250 and fixes the failing test. It also fixes some expired TODOs for the support of TS <4.4 which we've dropped. I can remove the renovate commit from this PR if needed

### CODE REVIEW

File: src/utils/object.ts [hunk 1]
```diff
@@ -174,19 +174,17 @@ type Flatten<T extends object> = object extends T ? object : {
  * @private
  */
 export function defaults(
-  objectIn: { [key: string]: any }, // TODO [2022-09-01]: key should be string | symbol once we drop support for TS 4.4
-  ...sources: Array<{ [key: string]: any }> // TODO [2022-09-01]: key should be string | symbol once we drop support for TS 4.4
+  objectIn: { [key: string | symbol]: any },
```
Comment (ephys) [inline src/utils/object.ts:177]:
```suggestion
  objectIn: { [key: PropertyKey]: any },
```

**Review by ephys — APPROVED** [review]
Other than that LGTM

File: src/utils/format.ts [hunk 2]
```diff
@@ -154,7 +151,7 @@ export function mapWhereFieldNames(where: Record<string | symbol, any>, Model: M
  * @returns All keys including operators
  * @private
  */
-export function getComplexKeys(obj: object): Array<string | symbol> {
+export function getComplexKeys(obj: object): PropertyKey[] {
```
Comment (ephys) [inline src/utils/format.ts:154]: Note that PropertyKey is `string | symbol | number`, but numbers are stringified when used as a key so this method would only get back `string | symbol`

**Review by ephys — CHANGES_REQUESTED** [review]
The rest is good, only this signature needs an update

File: src/utils/format.ts (hunk 2 above)
Comment (WikiRik) [reply src/utils/format.ts:154]: In other cases using PropertyKey is fine right?

Comment (WikiRik) [reply src/utils/format.ts]: This is fixed

Comment (ephys) [reply src/utils/format.ts]: Yep!

**Review by ephys — APPROVED** [review]

**Pull request merged**